lib/UnexpectedExpressMocker.js

```diff
@@ -0,0 +1,366 @@
+          }
+        }
+
+        // handle calling next() with a status code
+        if (typeof err === 'number') {
+          const statusCode = err;
+          err = new Error(`${statusCode}`);
+          err.statusCode = statusCode;
+          errorPassedToNext = err;
+        } else {
+          errorPassedToNext = err;
+        }
+
+        resolve();
+      });
+    })
+      .then(() => {
+        if (res.connection._writableState.corked > 0) {
+          // Wait for the connection to become uncorked before proceeding
+          const originalUncork = res.connection.uncork;
+          return new Promise(resolve => {
+            res.connection.uncork = function() {
+              const returnValue = originalUncork.apply(this, arguments);
+              if (res.connection._writableState.corked === 0) {
+                resolve();
+              }
+              return returnValue;
+            };
+          });
+        }
+      })
+      .then(() => {
+        Object.assign(context, {
+          req,
+          res,
+          metadata: {
+            strictAsync: isAsync,
+            errorPassedToNext: false,
+            isDestroyed,
+            requestDestroyed,
+            nextCalled: nextCalls.length > 0,
+            locals: res.locals,
+            url: req.url
+          },
+          httpRequest
+        });
+
+        let hasWrittenErrorStatusCode = false;
+        if (errorPassedToNext && !res.headersSent) {
+          hasWrittenErrorStatusCode = true;
+          let statusCode;
+          const statusProperty = determineStatusProperty(errorPassedToNext);
+          if (statusProperty) {
+            statusCode = errorPassedToNext[statusProperty];
+          } else {
+            statusCode = 500;
+          }
+          res.statusCode = statusCode;
+          res.writeHead(statusCode);
+        }
+
+        if (!res.headersSent) {
+          res.statusCode = 404;
+          // Make sure that the already set headers get flushed:
+          res.writeHead(404);
+        }
+
+        const httpResponse = (context.httpResponse = new messy.HttpResponse(
+          rawResponseChunks.length > 0
+            ? Buffer.concat(rawResponseChunks)
+            : res._header
+        ));
+        if (typeof httpResponse.rawBody === 'undefined') {
+          httpResponse.rawBody = Buffer.from([]);
+        }
+        httpResponse.statusCode = httpResponse.statusCode || res.statusCode;
+
+        if (errorPassedToNext) {
+          if (typeof expectedErrorPassedToNext !== 'undefined') {
+            if (expectedErrorPassedToNext === true) {
+              context.metadata.errorPassedToNext = true;
+            } else if (typeof expectedErrorPassedToNext === 'string') {
+              context.metadata.errorPassedToNext = errorPassedToNext.message;
+            } else {
+              context.metadata.errorPassedToNext = errorPassedToNext;
+            }
+          } else if (determineStatusProperty(errorPassedToNext)) {
```

Comment:
This chunk could do with some special attention. Error classification is reworked so the mocker always resolves but records a named error in cases where the assertion previously threw. To avoid behavioural change, when used via the assertion there are still rethrown: https://github.com/unexpectedjs/unexpected-express/blob/b358c3f1187d8e69710510d9f45c19fb9c626daf/lib/unexpectedExpress.js#L67